File: src/main/java/com/atsuishio/superbwarfare/datagen/DataGenerators.java
package com.atsuishio.superbwarfare.datagen;

import com.atsuishio.superbwarfare.Mod;
import net.minecraft.core.HolderLookup;
import net.minecraft.data.DataGenerator;
import net.minecraft.data.PackOutput;

import java.util.concurrent.CompletableFuture;

public class DataGenerators {

    public static void gatherData(DataGenerator generator, PackOutput packOutput, CompletableFuture<HolderLookup.Provider> lookupProvider) {
        generator.addProvider(true, ModLootTableProvider.create(packOutput));
        generator.addProvider(true, new ModRecipeProvider(packOutput));
        generator.addProvider(true, new ModBlockStateProvider(packOutput));
        generator.addProvider(true, new ModItemModelProvider(packOutput));
        ModBlockTagProvider tagProvider = generator.addProvider(true, new ModBlockTagProvider(packOutput, lookupProvider));
        generator.addProvider(true, new ModItemTagProvider(packOutput, lookupProvider, tagProvider.contentsGetter()));
        generator.addProvider(true, new ModEntityTypeTagProvider(packOutput, lookupProvider));
        generator.addProvider(true, new ModDamageTypeTagProvider(packOutput, lookupProvider));
        generator.addProvider(true, new ModAdvancementProvider(packOutput));
    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/datagen/DataGenerators.kt
package com.atsuishio.superbwarfare.datagen

import net.fabricmc.fabric.api.datagen.v1.DataGeneratorEntrypoint
import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator

class DataGenerators : DataGeneratorEntrypoint {
    override fun onInitializeDataGenerator(fabricDataGenerator: FabricDataGenerator) {
        val pack = fabricDataGenerator.createPack()

        pack.addProvider { output -> ModLootTableProvider.create(output) }
        pack.addProvider { output -> ModRecipeProvider(output) }
        pack.addProvider { output -> ModBlockStateProvider(output) }
        pack.addProvider { output -> ModItemModelProvider(output) }

        val tagProvider = pack.addProvider { output, registries -> ModBlockTagProvider(output, registries) }
        pack.addProvider { output, registries -> ModItemTagProvider(output, registries, tagProvider.contentsGetter()) }
        pack.addProvider { output, registries -> ModEntityTypeTagProvider(output, registries) }
        pack.addProvider { output, registries -> ModDamageTypeTagProvider(output, registries) }
        pack.addProvider { output -> ModAdvancementProvider(output) }
        pack.addProvider { output, registries -> ModPerkTagProvider(output, registries) }
        pack.addProvider { output -> ModWreckageLootProvider(output) }
    }
}
